parsl/executors/high_throughput/manager_selector.py

```diff
@@ -0,0 +1,22 @@
+import copy
+import random
+from typing import Dict, List
+
+from parsl.executors.high_throughput.manager_record import ManagerRecord
+
+
+class ManagerSelectorBase:
```

Comment:
have a look at how abstract base class is used elsewhere to force users to implement methods / prohibit instantiating classes that aren't subclassed as much as they need to.

this could force some errors when people are developing new managers to occur when the config object is created (at the point they define their configuration, early on in their run) rather than this raising a `NotImplementedError` inside the interchange code where it is harder to understand.

parsl.executors.base would give you an example of `ABCMeta` and `@abstractmethod`